lib/rubocop/cop/sorbet/signatures/signature_build_order.rb

```diff
@@ -70,6 +70,16 @@ def autocorrect(node)
 
         private
 
+        def node_with_index_sends(node)
+          # This is really dirty hack to reparse the current node with index send
+          # emitting enabled, which is necessary to unparse them back as index accessors.
+          emit_index_value = RuboCop::AST::Builder.emit_index
+          RuboCop::AST::Builder.emit_index = true
+          RuboCop::AST::ProcessedSource.new(node.source, target_ruby_version, processed_source.path).ast
```

Comment:
Instead of setting a global flag and use rubocop you can use `Unparser.parse` directly on the `node.source`. It runs (without setting global flags) the correct "modern" parser.

Reply:
We have been trying to not get a hard dependency on `Unparser` for the cop to work, but only enabled autocorrect if it existed. It seems that I was too hung up on that requirement and missed the fact that this is a correct phase operation so can depend on the existence of `Unparser`.

Thanks, I'll take a look at that.

Reply:
@paracycle Cool. There is also `Unparser.parser` which returns a "correctly" configured parser, possibly you can push that one into rubocop, so rubocop still controls the parsing process.

Also I could see myself actually supporting target ruby versions in unparser, but that would require funding. For `mutant` (which is the primary reason unparser exists) this is not required.

OT: Unparser recently got all "unique" dependencies removed, see https://rubygems.org/gems/unparser/versions/0.6.0, you may be able to reduce end user usage complexity in going for a hard dependency.

I'ts likely I can remove the `diff-lcs` dependency soon in factoring out `unparser-cli` from `unparser`. But this should not too much be an issue as `diff-lcs` is a very common dependency.

Reply:
Hmm, I can't quite get `Unparser.parser` to work, since Rubocop AST works by supplying a custom `Builder` to the parser object that creates special node types with extra methods and attributes. The call to `root_call`, which is a node matcher, fails if the supplied node is not a Rubocop AST node type. :(

Reply:
@paracycle I see, I wounder if you can instead supply a builder to rubocop that is defined like this? https://github.com/mbj/unparser/blob/3a459c8fbe117d3a78c1efe48c76ddfee175816f/lib/unparser.rb#L22-L31

But inheriting the rubocop builder? This way you do not set globals.

OT: I wounder what it would take to rubocop to use the modern AST format by default? You are de-facto sending it modern AST already, so the change rubocop side may be easier.

Reply:
🤯 Of course, that seems to work beautifully: https://github.com/Shopify/rubocop-sorbet/pull/51